CHANGES.rst

```diff
@@ -13,6 +13,11 @@ Deprecated
 * #11693: Support for old-style :file:`Makefile` and :file:`make.bat` output
   in :program:`sphinx-quickstart`, and the associated options :option:`!-M`,
   :option:`!-m`, :option:`!--no-use-make-mode`, and :option:`!--use-make-mode`.
+* #11285: Direct access to :attr:`!sphinx.testing.util.SphinxTestApp._status`
+  or :attr:`!sphinx.testing.util.SphinxTestApp._warning` is deprecated. Use
+  the public properties :attr:`!sphinx.testing.util.SphinxTestApp.status`
+  and :attr:`!sphinx.testing.util.SphinxTestApp.warning` instead.
```

Comment:
I don't see this mentioned in the linked issue thread - what's the reason to change these?

It also diverges a bit from the [main `Sphinx` class `._warning` attribute](https://github.com/sphinx-doc/sphinx/blob/23dd754f6375afb62e8af1e9e3b17d20b113fcc0/sphinx/application.py#L177-L180) - is that OK?

Reply:
I'm not worried about breaking downstream applications, but sometimes I build the projects in the `tests/roots` externally by calling `sphinx-build` directly on them so that I can more easily inspect/compare how the constructed content changes.  So that makes me cautious about any changes that might mean that `SphinxTestApp` behaves differently to `sphinx.application.Sphinx`.

(it's OK though - it's a small detail and I think review/understanding of the other aspects is more important)

Reply:
> I build the projects in the tests/roots externally by calling sphinx-build

The command-line is not affected by that. However, it's real that the SphinxTestApp (current) implementation differs from its parent class (which I really found it weird).